projects/mysadna/src/app/services/encoding.service.ts

```diff
@@ -1,8 +1,6 @@
 import { Injectable } from '@angular/core';
 
-@Injectable({
-  providedIn: 'root'
```

Comment:
Why do you use providedIn root there?

Reply:
Auto generated code.
I didn't know I should remove it.

Reply:
I got it.
You don't have to remove it, but I don't see any sense to use providedIn, if you don't understand purpose of the code.